### ISSUE TITLE
Update PHPStan issues in some EventSubscribers and CmsProtectionSubscriber

### ISSUE THREAD

**Issue by bryanpizzillo** [NCIOCPL/cgov-digital-platform#4026]
### Issue description
> **ESTIMATE** TBD

The following EventSubscribers need to be updated. They all should just be changing out RequestEvent.

* `cgov_core/src/EventSubscriber/TranslationAvailableSubscriber.php`
   * The way Drupal handles language urls is:
      * If the URL starts with `/espanol` it is Spanish, otherwise it is English. This determines what language the User Interface is in.
      * Next, it looks at the rest of the URL `/node/XXX` or `/media/XXX` to determine what content to load.
      * It then attempts to get the content for the User Interface language. If the content does not exist in that language, IT SHOWS THE ORIGINAL LANGUAGE.
      * This bit makes it so that if you ask for a URL that the node or media is not translated into, then you get a Page Not Found.
      * See #296 for steps to test / more information.
* `cgov_file/src/EventSubscriber/DirectFileDownloadSubscriber.php`
   * If you request a pretty url, or `/media/NNN` url, for a cgov_file, then you should download the file. Drupal would normally draw a page with a link to the file or something.
   * See #1166

For CmsProtectionSubscriber:
Basically this is what makes sure that going to www-cms.cancer.gov/foo as an unauthenticated user redirects to the login page. See #2014 for original requirements.

**Comment by navsunka** [comment]
Testing DirectFileDownloadSubscriber in ACSF

- Go to /admin/content/media and replace existing file (.json) with empty json file.
- Refresh the page and we should see the new file empty JSON.
- Also, Click on the file and inspect network tab, we should see the Edge-Cache-Tag value.
- we should check by replacing different file formats as well.

**Comment by alinai** [comment]
> @navsunka to confirm the test steps. cc: @KateMashkinaNIH , @santwuma

@navsunka has confirmed that test step involves matching the above scenario with cms-dev.

**Comment by alinai** [comment]
@navsunka , I have tested this ticket in the following way: 
Test Steps:
1) Created a file
2) Uploaded a file (e.g. json) with content
3) Published the file media type
3) Edited the file created in Step#1
4) Replaced the file (.json) with an empty/blank (e.g. json) file
5) Published 
6) Clicked on the Title of the edited and published (.json) file (from Step#4) and observed the different download behaviors
7) Repeated step#1v to step#6 for different file formats

@navsunka , Please confirm if the below test results are expected: 
**.json file:**
File with content: On clicking the .json file opens and user can see the content on the same browser tab in both cms-dev and cms-test
Empty file: On Clicking the .json file downloads with a c: drive logo and when opened, I can see that it is a 0KB file in both  cms-dev and cms-test.

![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/a9b32fd4-59b8-40ca-b3f6-75beb64e803a)


**.xlsx file:**
File with content: On clicking, the .xlsx file gets downloaded in both cms-dev and cms-test with proper .xlsx logo appearing beside filename (See the screenshot#1)
Empty file: On clicking the .xlsx file downloads with a c: drive logo instead of .xlsx logo in both cms-dev and cms-test (screenshot#2)
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/6ccd1f18-0f47-4589-ac4b-366b28c19c4d)

![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/0cd9ff21-ace6-4e00-aafe-1a6f53f19628)

**.csv file**
File with content: On clicking, the .csv file opens and user can see the content on the same browser tab in both cms-dev and cms-test
Empty file: On clicking the .csv file downloads with a c: drive logo instead of .csv logo in both cms-dev and cms-test 

**.docx file:**
File with content: On clicking, the .docx file gets downloaded in both cms-dev and cms-test with proper .docx logo appearing beside filename (See the screenshot#1)
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/eefecfc6-29e4-487c-8550-9520d9ffe070)

Empty file: On clicking, the .docx file downloads with a c: drive logo instead of .docx logo in both cms-dev and cms-test (screenshot#2)
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/0de52ead-1a3b-4dac-899d-ded2a3bd08fe)

**.pptx:**
File with content: On clicking, the .pptx file gets downloaded in both cms-dev and cms-test with proper .pptx logo appearing beside filename (See the screenshot)
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/6324cedf-49bf-4d2d-9c56-e89ee541f66d)
Empty file: On Clicking, the .pptx file downloads with a c: drive logo in both  cms-dev and cms-test.
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/3008e104-7de3-4228-aa81-1230e5ca81ff)

**.pdf**
File with content: On clicking, the .pdf file opens and user can see the content on the same browser tab in both cms-dev and cms-test
Empty file: On clicking, the .pdf file opens and user can see the blank page on the same browser tab in both cms-dev and cms-test

**.rtf:** 
File with content: On clicking, the .rtf file gets downloaded with a c: drive logo  in both cms-dev and cms-test (See the screenshot)
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/91804dda-2bf1-44d4-ae01-75bd54931d6c)
Empty file: On clicking the .rtf file downloads with a c: drive logo in both cms-dev and cms-test 

.zip:
File with content: On clicking, the .zip file gets downloaded in both cms-dev and cms-test with proper .zip logo appearing beside filename (See the screenshot)
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/8fd98430-109d-48b2-9fdc-9a085de4f70d)
Empty File: On clicking, the .zip file downloads with proper logo appearing beside the file name
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/3d9d1331-6cd0-4635-9ac6-5a1fb6b2c01b)

**.doc**
File with content: On clicking, the .doc file gets downloaded in both cms-dev and cms-test with proper .doc logo appearing beside filename (See the screenshot#1)
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/b7274c90-22ed-4bd6-8309-701942155691)

Empty file:  On clicking, the .doc file downloads with .doc logo in both cms-dev and cms-test 
![image](https://github.com/NCIOCPL/cgov-digital-platform/assets/28455973/27f01d8a-75da-4433-957d-4f7d86373a44)

Note: For a particular file type (for example .docx), both cms-dev and cms-test behaves in the same way when we click on them. But the file logo changes to c:drive logo when empty file is downloaded for some file types (e.g .docx) in both cms-dev and cms-test. Is it expected?

cc: @KateMashkinaNIH , @santwuma , @navsunka

**Comment by naveen-kumar2** [comment]
Behavior is same in both environments, looks good to me. Thanks.

**Comment by KateMashkinaNIH** [comment]
TranslationAvailableSubscriber and CmsProtectionSubscriber - expected behavior
